fiftyone/core/collections.py

```diff
@@ -594,13 +594,23 @@ def stats(self, include_media=False, compressed=False):
 
     def _get_samples_bytes(self):
         """Computes the total size of the sample documents in the collection."""
+
+        # Impl note: We *could* do group directly without projecting $bsonSize
+        #   first, however we found in Mongo 5.2 a new execution engine
+        #   (Slot Based Execution) optimizes this away and incorrectly sums
+        #   the full document size, thus ignoring selected fields. A ticket
+        #   has been opened to resolve this bug.
+        #   https://jira.mongodb.org/browse/SERVER-75267
```

Comment:
Did some reading on slot based execution, and it's possible this is the expected behavior.

Previous engine outputted documents at each stage, so $$ROOT would refer to the document of the projected fields in that stage, but with slot based, the $$ROOT is now the original document (no intermediate document created)

> Hey! Query Execution engineer here, my team and I are working on SBE. Currently, query execution engine works in terms of documents. For example, in the aggregation language, every stage produces a document to be consumed by the next stage. SBE goes one step lower in abstraction layers and works in terms of values. Aggregation stages executed through SBE pass values to each other directly, without extra document materialization. This allows us to avoid many unnecessary computations and execute queries faster. As of now, we only use SBE for certain queries with $group and $lookup, but the intent is to use it across the board in the future.
https://www.reddit.com/r/mongodb/comments/v7uaso/what_does_the_slotbased_query_execution_engine_do/

Reply:
An execution engine should not change functionality unless documented (it is not), so either way it is a bug IMO.
They may come back (if at all) and say yeah not gonna change that. In which case their documented way of getting total document size is just broken